### PR TITLE
New component: TextAction

### PR DESCRIPTION
**Pull request by leirbag95**
This pull request adds a new component called **TextAction**, which allows the user to create a text with an action next to it. This component is useful when you want to display a text and provide an action that the user can take related to the text.
You can provide the `action` prop which is used to specify the action to be taken when the user clicks on the action button.

I have tested the TextAction component locally and it works as expected with Copy to Clipboard function. 

Please review and let me know if you have any feedback or suggestions for improvements.

image

### CODE REVIEW

File: src/components/TextAction/cmp.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+          color={title.color}
+          value={title.value}
+        ></Title>
+        <div
+          style={{
+            display: 'flex',
+            alignItems: 'center',
+            justifyContent: position,
+          }}
+        >
```
Comment (BjrInt) [inline src/components/TextAction/cmp.tsx:103]: This is not needed, keep it simple :wink:

File: src/components/TextAction/types.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import { TypoKind } from '../../themes'
+import { IconName } from '@fortawesome/fontawesome-svg-core'
+
+export type TitlePositionProps = 'left' | 'center' | 'right'
+
+export type TitleProps = {
+  color: string
+  value: string
+  size: number
+}
```
Comment (BjrInt) [inline src/components/TextAction/types.ts:11]: You can remove this also.

File: src/components/TextAction/cmp.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  position,
+  type = 'body1',
+  title,
+  color,
+  icon,
+}: TextActionProps) => {
+  const [iconName, setIconName] = useState<string>(icon)
+  const [iconColor, setIconColor] = useState<string>(color)
+  const [showIcon, setShowIcon] = useState<boolean>(true)
+  const [, copyToClipboard] = useClipboard(content)
```
Comment (BjrInt) [inline src/components/TextAction/cmp.tsx:61]: Thanks for creating the hook, but since this is intended as a generic component, could you allow to pass any function as a callback prop, instead of calling this by default ?

File: src/components/TextAction/cmp.tsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+      }, 2200)
+    }, duration)
+  }
+  return (
+    <StyledTextAction
+      {...{
+        type,
+        position,
+        color,
+      }}
```
Comment (BjrInt) [inline src/components/TextAction/cmp.tsx:89]: This returns a Typescript error since StyledTextAction requires several other props.

File: src/components/TextAction/cmp.tsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+export const TextAction = ({
+  size,
+  content,
+  position,
+  type = 'body1',
+  title,
+  color,
+  icon,
+}: TextActionProps) => {
+  const [iconName, setIconName] = useState<string>(icon)
```
Comment (BjrInt) [inline src/components/TextAction/cmp.tsx:58]: iconName is not a string (please import the type from FontAwesome)

File: src/components/TextAction/cmp.tsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+            kind={'neon'}
+            size={'regular'}
+            style={{ padding: '0 0 0 0.5rem' }}
+          >
+            <Transition in={showIcon} timeout={duration}>
+              {(state: string | number) => (
+                <div
+                  style={{
+                    ...defaultStyle,
+                    ...transitionStyles[state],
```
Comment (BjrInt) [inline src/components/TextAction/cmp.tsx:119]: You cannot use a generic string or number to access the key of an object (maybe a keyof type would help here?)

**Review by BjrInt — CHANGES_REQUESTED** [review]
Hey,

Thanks for the PR. 
Can you please keep this component minimal, just the text, the icon, and a callback as a prop. There is some unecessary bloat (that's also causing type errors) that could easily be fixed by trimming the code. 

TL;DR: Just keep this (without title, align, type, ...) 
image

**Comment by leirbag95** [comment]
> Hey,
> 
> Thanks for the PR. Can you please keep this component minimal, just the text, the icon, and a callback as a prop. There is some unecessary bloat (that's also causing type errors) that could easily be fixed by trimming the code.
> 
> TL;DR: Just keep this (without title, align, type, ...) image

Ok I'm going to fix it today

**Review by BjrInt — APPROVED** [review]
